### PR TITLE
sds1104xe: add offsetdac

### PR DESCRIPTION
**Pull request by tmbinc**
Add offsetdac&offsetmux, which are used on SDS1104X-E to set the offset voltage for each of the 4 channels with a single high-precision DAC and an analog MUX.

The state machine continuously refreshes all 8 inputs by setting the DAC, setting the Mux, and then relying on the external analog sample&hold logic to keep the value. To prevent drift, this is done with a relatively high rate.

### CODE REVIEW

File: litex_boards/platforms/sds1104xe.py [hunk 1]
```diff
@@ -66,36 +66,50 @@
...
+        Subsignal("nSYNC",      Pins("J15")),
+        IOStandard("LVCMOS15"),
+    ),
+
     # DDR3 SDRAM
     ("ddram", 0,
         Subsignal("a", Pins(
             "J21 K18 J18 R16 P16 T18 R18 T19",
             "R19 P18 P17 P15 N15"),
-            IOStandard("SSTL135")),
```
Comment (tmbinc) [inline litex_boards/platforms/sds1104xe.py:74]: Some/all of the DDR pins are shared with LVCMOS15 pins. We need to double check what the IO standard for the DDR really is.

File: litex_boards/targets/sds1104xe.py [hunk 2]
```diff
@@ -36,6 +36,8 @@
 
 from liteeth.phy.mii import LiteEthPHYMII
 
+from sds1104xe_offsetdac import OffsetDac
```
Comment (tmbinc) [inline litex_boards/targets/sds1104xe.py:39]: Where do we put more complex per-board logic? This component can likely be shared on similar hardware (such as the SDS1202X-E, the 2-channel variant).

File: litex_boards/targets/sds1104xe_offsetdac.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    def __init__(self, offsetdac_pads, offsetmux_pads):
+        self._status = csr.CSRStatus(32, name='status', reset=0x0ffdac)
+        self.spi = Record(SPI)
+        self.mux = Record(MUX)
+
+        self.comb += self.spi.connect(offsetdac_pads)
+        self.comb += self.mux.connect(offsetmux_pads)
+
+        self._control = csr.CSRStorage(32, name='control') # DAC PD control
+        self._clkdiv = csr.CSRStorage(32, name='clkdiv', reset=326) # roughly matches original timing
```
Comment (tmbinc) [inline litex_boards/targets/sds1104xe_offsetdac.py:25]: This needs to be fixed corresponding to the sysclk

**Comment by enjoy-digital** [comment]
Thanks @tmbinc, as suggested on IRC, LiteX-Boards is maybe not a right place for custom peripherals, maybe we could create a project under https://github.com/360nosc0pe for the specific scope part? And just try to base it on LiteX-Boards to avoid duplicating base component support (DRAM, Ethernet, etc...).

**Comment by tmbinc** [comment]
That makes sense - we'll maintain this under 360nosc0pe for now.

**Pull request closed without merging**